quickwit-indexing/src/actors/merge_planner.rs

```diff
@@ -66,46 +70,88 @@ impl Handler<NewSplits> for MergePlanner {
         message: NewSplits,
         ctx: &ActorContext<Self>,
     ) -> Result<(), ActorExitStatus> {
-        let mut has_new_young_split = false;
-        for split in message.new_splits {
-            if self.merge_policy.is_mature(&split) {
-                info!(split_id=%split.split_id(), num_docs=split.num_docs, size_in_bytes=split.uncompressed_docs_size_in_bytes, "mature-split");
-                continue;
-            }
-            self.young_splits.push(split);
-            has_new_young_split = true;
-        }
-        if has_new_young_split {
-            self.send_operations(ctx).await?;
+        let mut target_partition_ids = Vec::new();
+
+        let partitioned_new_young_splits = message
+            .new_splits
+            .into_iter()
+            .filter(|split| {
+                !self.merge_policy.is_mature(split) || {
```

Comment:
I see the problem you are solving, but I'm not a big fan of the usage `||`

Reply:
Yes, I will use an "explanatory" variable instead.